server/aesdsocket.c

```diff
@@ -0,0 +1,258 @@
+/*
+file: server.c
+description: TCP/IP socket server code 
+Author: Jahnavi Pinnamaneni; japi8358@colorado.edu
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <syslog.h>
+
+#include <signal.h>
+int server_socket;       //socket_fd
+int client_socket;       //client_fd
+
+void daemonise_process();
+
+/*
+Signal handler to gracefully terminate when signals SIGINT and SIGTERM arrive
+parameter: signal number
+*/
+void graceful_shutdown(int signo)
+{
+    syslog(LOG_DEBUG, "CAUGHT SIGNAL, EXITING GRACEFULLY \n");
+    remove("/var/tmp/aesdsocketdata");
```

Comment:
@Jahnavi-pinnamaneni is it safe to call async-signal unsafe functions like syslog, remove here in signal handler?